### ISSUE TITLE
Error when showing the field on a saved Jelly_Field_Timestamp

### ISSUE THREAD

**Issue by vitch** [jonathangeiger/kohana-jelly#122]
I'm not sure if this is a bug or a mistake in my implementation but this is the issue I am having:

Create a model with a type of field of type Field_Timestamp e.g.

```
'date_published' => new Field_Timestamp(
    array(
        'pretty_format' => 'Y-m-d',
        'format' => 'Y-m-d',
        'default' => time(),
    )
),
```

Now create a form which uses `$model->input($field->name)` to output a text field for this field. Then submit your form which should save the model. When the page tries to output the field in the same request as the save I get an error:

```
ErrorException [ Notice ]: A non well formed numeric value encountered
MODPATH/jelly/views/jelly/field/timestamp.php [ 1 ]
```

I think it's because when you call save() on the model the value is converted into its DB representation which isn't what is expected when you call $model->input...

**Comment by vitch** [comment]
Possible fix in vitch@e90ea9ef6c47ce00b3897164454cd79d6f0d04dc but not sure if this is the intended approach/ use?

**Comment by paulchubatyy** [comment]
Please check out http://github.com/jonathangeiger/kohana-jelly/issues/closed#issue/107 seems we were solving the same issue.

**Comment by vitch** [comment]
Thanks for the info but I think the problem is slightly different. It is interesting to see that Jelly assumes that the underlying field for a Timestamp is an int - this doesn't make much sense to me... MySQL's datetime datatype is much more suited and it means that you can very easily e.g. retrieve all records from a particular month (e.g. "WHERE DATE_FORMAT(field, '%Y-%m') = '2010-06'") etc

**Comment by glamorous** [comment]
Everyone who uses this Jelly's field date has to use a INT, if you wan't this, you can make your own field. The reason why should use a INT is that your date has the same format in every kind of database (MySQL, SQLite, ...)

**Comment by jonathangeiger** [comment]
glamorous' explanation is not entirely correct. When working with the field in PHP, an int will be returned because most of the PHP date functions expect a UNIX timestamp. However, you can set the 'format' option on a Field_Timestamp and the field will automatically convert the current timestamp back to the format specified before saving. It seems you already understand this though.

Can you post the entire backtrace?

**Comment by vitch** [comment]
Thanks for the response. Yes - I had already discovered the format option and am using it.

I have posted the entire stack trace here:

http://www.kelvinluck.com/assets/temp/issue_122.html

(note that I just reverted my project repo to a point in the past to get the error as I have worked around it in my code)

Let me know if there is any more information I can provide to help you get to the bottom of the problem. It may be down to me doing something stupid because I'm pretty new to Jelly...

**Comment by jonathangeiger** [comment]
Hmm. Can you let me know where your model's coming from by posting a bit of the code. I can see you're passing it to a view, but what's the code you're using for retrieving it and what processing are you doing before passing it to the view?

The problem is that in your Model_News the value for 'date_published' is '2010-04-06' when it should have been converted to an int (1270530000) by then. I just can't figure out why that hasn't happened.

**Comment by vitch** [comment]
Once I've passed my current deadline I'll try to revert the changes and re-create the problem. But basically the flow is:
- Create model
- If postback then update values (for this field that would include set()ing the value to the string submitted by the user.
- Render the form by looping over the model and outputting each field.

I suspect the problem is because when you set() the value to a string it is storing the string internally rather than converting it back to the int representation or something?

**Comment by jonathangeiger** [comment]
Are you set()ting on the model or the builder when you update? If you're doing

```
$model->set('date_published', $value);
```

it should convert it fine. However, this won't work:

```
Jelly::update('news')->set('date_published', $value)->execute();
```

**Comment by jonathangeiger** [comment]
Just figured it out. Will get back to you.

**Comment by jonathangeiger** [comment]
Closed by 3745df0069f80d3d19231cc5363735e543d4f500. Thanks vitch!

**Comment by vitch** [comment]
Thanks - that works for me and enabled me to remove my workaround. I am running into a very weird issue with certain dates though. Not sure if it is related or if I should open a new ticket... 

With format and pretty_format both set to 'Y-m-d' it works fine most of the time. But certain dates get changed into other dates. e.g. If I try and save '2009-10-03' the date comes back from the database as '4400-06-10' - any ideas?

**Comment by jonathangeiger** [comment]
That's strange. Any code or backtraces you can post?

**Comment by vitch** [comment]
Fixed in vitch/kohana-jelly@3826e49a3c49104ac4820a0a05984ea6653c408e - for an illustration of the problem see the following code:

```
$dates = array('2009-10-05', '2009-10-03');
foreach ($dates as $my_date) {
    $t = strtotime($my_date);
    echo $my_date . ' = ' . $t . ' = ' . date('Y-m-d', $t);
    echo '<br />';
    echo strtotime($t); // strtotime($t) !== FALSE!!!!
    echo '<br />';
}
```

Which outputs:

```
2009-10-05 = 1254697200 = 2009-10-05

2009-10-03 = 1254524400 = 2009-10-03
76697412892
```

The code in Jelly_Field_Timestamp was calling strtotime on a number and as you can see, this can lead to unexpected results.
